src/ebonite/core/objects/requirements.py

```diff
@@ -127,7 +143,13 @@ def source(self) -> str:
         """
         Source code of this requirement
         """
-        return CustomRequirement.decompress(self.source64zip)
+        if not self.is_package:
+            return CustomRequirement.decompress(self.source64zip)
```

Comment:
I propose to raise some exception here and in the next method instead of silently returning None.